milli/src/search/mod.rs

```diff
@@ -169,6 +176,19 @@ impl<'a> Search<'a> {
             ctx.attributes_to_search_on(searchable_attributes)?;
         }
 
+        if let Some(distinct) = &self.distinct {
+            let filterable_fields = ctx.index.filterable_fields(ctx.txn)?;
+            if !filterable_fields.contains(distinct) {
+                let (valid_fields, hidden_fields) =
+                    ctx.index.remove_hidden_fields(ctx.txn, filterable_fields)?;
+                return Err(Error::UserError(UserError::InvalidDistinctAttribute {
+                    field: distinct.clone(),
+                    valid_fields,
+                    hidden_fields,
+                }));
```

Comment:
Hey, I think we're missing something here.
I pushed a test highlighting that it doesn't work with nested fields which I don't think is expected? 🤔 

Shouldn't we check if the filterable fields are being faceted by the filterable fields instead of a simple `contains`?